IIA-2912 Semantic Title Not Updated Until Semantic Window Is Reopened

bhharsh13 opened the pull request:
Fixed the Semantic Title display -https://ikmdev.atlassian.net/browse/IIA-2912

Summary of Changes:
Added a listener to refComponentProp, wrap the UI update inside Platform.runLater.

**Before Change:**
<img width="1412" height="1300" alt="image-20250917-140527" src="https://github.com/user-attachments/assets/9e180f94-a1ea-49ef-89bc-7bb42ed0bf98" />

**After Change:**
<img width="880" height="389" alt="New Semantic title" src="https://github.com/user-attachments/assets/6f4dd5c9-4037-4644-966c-95518351fe01" />
<img width="877" height="362" alt="After adding Reference Component" src="https://github.com/user-attachments/assets/032379a5-d32e-4470-9fb3-b2f32a2b406a" />

<img width="520" height="376" alt="Semantic title updated" src="https://github.com/user-attachments/assets/a29953e9-2c1a-491f-b7db-dae66fb1608c" />

jdsmithsos reviewed (commented):
I just made the same kind of fix in Pattern window, but for the Identifier not being set after creation.  What I did was check the edit mode change to EDIT, then set the identifier.
See my PR https://github.com/ikmdev/komet/pull/770.

Does your change follow the same pattern that Pattern and Concept windows use to set the title after creation?

(Honestly I'm not sure what Concept and Semantic do to set the Identifier after creation, I didn't even look, but probably should now since I brought this up)

bhharsh13 commented:
Addressed the code review comments. 
Validated the changes on local and looks good.
<img width="608" height="411" alt="image" src="https://github.com/user-attachments/assets/09d584b2-f962-41f2-8771-4bef4a98b7f4" />

swaroopsalvi commented on kview/src/main/java/dev/ikm/komet/kview/mvvm/view/genediting/GenEditingDetailsController.java:
```diff
@@ -629,6 +629,7 @@ private void setupReferenceComponentUI() {
                 case ConceptEntity ignored -> {
                     refType = "Concept";
                     description = refComponent2.description();
+                    updateSemanticForPatternInfo();
```
move this call to the end for the function after the 

`` referenceComponent.setTitle(refType);
            referenceComponent.setValue(componentItem);
``

bhharsh13 replied:
Made the changes as suggested.
<img width="881" height="464" alt="image" src="https://github.com/user-attachments/assets/1194c1c4-1564-4192-850c-158e61a0232e" />

swaroopsalvi reviewed (approved):
looks good Approved.
